ducktape/tests/loader.py

```diff
@@ -454,23 +457,45 @@ def _load_test_suite_files(self, test_suite_files):
             all_contexts.update(self._load_test_suite(**suite))
         return all_contexts
 
-    def _read_test_suite_from_file(self, test_suite_file_path):
-        suites = list()
-        test_suite_file_path = os.path.abspath(test_suite_file_path)
-        if not os.path.exists(test_suite_file_path) or not os.path.isfile(test_suite_file_path):
-            raise LoaderException("Test suite does not exist or is a directory: " + test_suite_file_path)
-
-        with open(test_suite_file_path) as fp:
+    def _load_suite(self, suite_file_path):
+        with open(suite_file_path) as fp:
             try:
                 file_content = yaml.load(fp, Loader=yaml.FullLoader)
             except Exception as e:
-                raise LoaderException("Failed to load test suite from file: " + test_suite_file_path, e)
-
-            if not file_content:
-                raise LoaderException("Test suite file is empty: " + test_suite_file_path)
-            if not isinstance(file_content, dict):
-                raise LoaderException("Malformed test suite file: " + test_suite_file_path)
-
+                raise LoaderException("Failed to load test suite from file: " + suite_file_path, e)
+
+        if not file_content:
+            raise LoaderException("Test suite file is empty: " + suite_file_path)
+        if not isinstance(file_content, dict):
+            raise LoaderException("Malformed test suite file: " + suite_file_path)
+
+        for suite_name, suite_content in file_content.items():
+            if not suite_content:
+                raise LoaderException("Empty test suite " + suite_name + " in " + suite_file_path)
+        return file_content
+
+    def _read_test_suite_from_file(self, root_suite_file_path):
+        suites = []
+        root_suite_file_path = os.path.abspath(root_suite_file_path)
```

Comment:
It is possible to pass multiple test suite files into the loader, so this can be optimized a bit by adding all the suite files in the stack at once.
I would also separate these two steps - one step should walk the graph and build a set of unique files to load; and a second one would simply load them. This is not critical, but might make the code simpler and functions shorter.

Reply:
the issue with that is to walk the graph it needs to load the file. (hence i cache the loaded files to be accessed later) 
I will go ahead and load all files at once good catch.

Reply:
Gotcha, yeah, that makes sense. 
It can still be done in two steps - one reads the files and only parses the import statement, thus filtering duplicate files; another reads test suites from the loaded files - but in this case the benefits of splitting them are questionable :)